dapps/marketplace/src/hoc/withFingerprint.js

```diff
@@ -2,7 +2,10 @@ import React, { useState, useEffect } from 'react'
 import Fingerprint2 from 'fingerprintjs2'
 import memoize from 'lodash/memoize'
 
-let cachedFingerprintData
+import withWallet from './withWallet'
+
+let cachedFingerprintData, cachedUUID
+const uuidKey = 'rewards_uuid'
```

Comment:
Great naming. uuid is a perfect name for this id.

Reply:
😊